added forgot password feature

mansiPaTeLL opened the pull request:
@holtzmak @MummenRider @SanaKhan1997 Please review this PR. Now you have to sign up using legit email address if you want to reset your password.

mansiPaTeLL committed:
added forgot password feature

holtzmak commented on app/lib/core/view_models/forgot_password_view_model.dart:
```diff
@@ -0,0 +1,35 @@
+      locator<AuthenticationService>();
+  final NavigationService _navigationService = locator<NavigationService>();
+
+  void navigateToSignInScreen() =>
```
- [x] If we're coming from the SignInScreen to this screen, we should just `navigateBackTo` and not nav and replace. 

Nav and replace is meant for parallel screens. This screen is more meant for "sitting on top"

holtzmak commented on app/lib/core/view_models/sign_in_view_model.dart:
```diff
@@ -18,6 +19,9 @@ class SignInViewModel extends BaseViewModel {
       _navigationService.navigateBackUntil(WelcomeScreen.route);
 
+  void navigateToForgotPasswordScreen() =>
+      _navigationService.navigateAndReplace(ForgotPasswordScreen.route);
```
- [x] Should be `navigateTo`

holtzmak reviewed (changes requested):
Minor navigation changes. Thanks!

mansiPaTeLL commented:
@holtzmak @MummenRider @SanaKhan1997  
![Screen Shot 2021-03-12 at 1 22 13 PM](https://user-images.githubusercontent.com/44125953/110988525-455b0e00-8336-11eb-825b-bb3f993f5aa7.png)
![Screen Shot 2021-03-12 at 1 22 34 PM](https://user-images.githubusercontent.com/44125953/110988535-49872b80-8336-11eb-882f-a6f6ac91c51e.png)
![Screen Shot 2021-03-12 at 1 22 58 PM](https://user-images.githubusercontent.com/44125953/110988546-4d1ab280-8336-11eb-9717-c804ee3694b7.png)
![Screen Shot 2021-03-12 at 1 23 13 PM](https://user-images.githubusercontent.com/44125953/110988557-4f7d0c80-8336-11eb-8678-7b4011901929.png)
![Screen Shot 2021-03-12 at 1 23 28 PM](https://user-images.githubusercontent.com/44125953/110988600-5f94ec00-8336-11eb-8cc0-fa07ea60334f.png)

SanaKhan1997 committed:
Rebbase changes

mansiPaTeLL committed:
added minor changes & gui

mansiPaTeLL committed:
added minor changes

mansiPaTeLL committed:
removed unused packages

mansiPaTeLL committed:
removed unused packages

mansiPaTeLL committed:
removed unused packages

holtzmak committed:
Update navigation to login screen from email confirmation screen

Minor GUI updates. Removed HexColor so we use named colours. Centered
the app welcome screen text.

holtzmak reviewed (approved):
Hi Mansi, I played around with your changes on my machine. I made a necessary navigation change to the email confirmation screen, and then added some GUI changes that were bothering me too. Sorry for piggybacking on your PR.

Everything looks good now!
